File: libs/maia-ui-workspace/projects/maia-elasticsearch/src/lib/elastic-indices/models/EsIndexStateDto.ts
import {ManagedEsIndexInfoDto} from './ManagedEsIndexInfoDto';
import {EsIndexHealthDto} from './EsIndexHealthDto';


export class EsIndexStateDto {
    exists!: boolean;
    health?: EsIndexHealthDto;
    indexName!: string;
    managedIndexInfo?: ManagedEsIndexInfoDto;
}

File: libs/maia-elasticsearch-parent/maia-elasticsearch/src/main/kotlin/org/maiaframework/elasticsearch/index/model/EsIndexStateDto.kt
package org.maiaframework.elasticsearch.index.model

import org.maiaframework.elasticsearch.index.EsIndexName


data class EsIndexStateDto(
    val indexName: EsIndexName,
    val summary: ManagedEsIndexSummaryDto?,
    val health: EsIndexHealthDto?
) {


    val indexExists = this.health != null


}
